ssc/cmod_wavefile.cpp

```diff
@@ -60,100 +78,255 @@ class cm_wave_file_reader : public compute_module
+            ssc_number_t* month = allocate("month", numberRecords);
+            std::vector<ssc_number_t> timecheck(numberRecords);
+            timecheck[0] = 0;
+            ssc_number_t* wave_heights = allocate("wave_significant_height", numberRecords);
+            ssc_number_t* wave_periods = allocate("wave_energy_period", numberRecords);
+            for (size_t r = 0; r < numberRecords; r++) {
+                getline(ifs, buf);
+                values.clear();
+                values = split(buf);
+                
+                if (r == 0) {
+                    //value_0 = split(buf);
+                    hour0 = (ssc_number_t)std::stod(values[hour_index]);
+                }
+                if (r == 1) {
+                    //value_1 = split(buf);
+                    hour1 = (ssc_number_t)std::stod(values[hour_index]);
+                    hourdiff = hour1 - hour0;
+                }
+                month[r] = (ssc_number_t)std::stod(values[month_index]);
+                timecheck[r] = (ssc_number_t)std::stod(values[hour_index]);
+                if (r > 0) {
+                    if (timecheck[r] - timecheck[r - 1] != hourdiff && timecheck[r] != 0) {
+                        throw exec_error("wave_file_reader", "Time steps are nonuniform");
+                        timecheck[r] = 999;
+                    }
+                }
+                p_year[r] = (ssc_number_t)std::stod(values[year_index]);
+                p_hour[r] = (ssc_number_t)std::stod(values[hour_index]);
+                p_day[r] = (ssc_number_t)std::stod(values[day_index]);
+                p_minute[r] = (ssc_number_t)std::stod(values[minute_index]);
+                wave_heights[r] = (ssc_number_t)std::stod(values[height_index]);
+                wave_periods[r] = (ssc_number_t)std::stod(values[period_index]);
+
+                //Make JPD from time series data
+
+                ts_significant_wave_height = wave_heights[r];
+                ts_energy_period = wave_periods[r];
+                for (size_t j = 0; j < 21; j++) {
+
+                    if (r == 0) mat[(j + 1) * ncols] = (0.25 + j * 0.5);
+                    if (abs(ts_significant_wave_height - (0.25 + (j - 1) * 0.5)) <= 0.25 && j != 0) {
+                        sig_wave_height_index = j;
+                        if (r != 0) break;
+                    }
+                }
+                for (size_t m = 0; m < 22; m++) {
+                    if (r == 0) mat[m] = m - 0.5;
+                    if (abs(ts_energy_period - (0.5 + (m - 1))) <= 0.5 && m != 0) {
+                        energy_period_index = m;
+                        if (r != 0) break;
+                    }
+
+
+                }
+
+
+                //mat[sig_wave_height_index * ncols + energy_period_index] = mat[sig_wave_height_index * ncols + energy_period_index] + 1 / 2920 * 100;
+                mat[sig_wave_height_index * ncols + energy_period_index] += 0.0342465753;
```

Comment:
does the increment value to the PDF depend on the number of records or time steps or anything?